node/peer.js

```diff
@@ -143,16 +145,19 @@ TChannelPeer.prototype.request = function peerRequest(options) {
     var req = self.connect().request(options);
 
     self.state.onRequest(req);
+    self.pending++;
 
     req.errorEvent.on(onError);
     req.responseEvent.on(onResponse);
 
     function onError(err) {
         self.state.onRequestError(err);
+        self.pending--;
```

Comment:
we probably want to clamp at zero, and afford a stat for when it tries to go < 0